homeassistant/components/zha/device_action.py

```diff
@@ -0,0 +1,65 @@
+"""Provides device actions for ZHA devices."""
+from typing import List
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_DEVICE_ID, CONF_DOMAIN, CONF_TYPE
+from homeassistant.core import Context, HomeAssistant
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers import service
+from homeassistant.helpers.typing import ConfigType, TemplateVarsType
+
+from . import DOMAIN
+from .api import SERVICE_WARNING_DEVICE_SQUAWK, SERVICE_WARNING_DEVICE_WARN
+from .core.const import CHANNEL_IAS_WD
+from .core.helpers import async_get_zha_device
+
+ACTION_SQUAWK = "squawk"
+ACTION_WARN = "warn"
+ATTR_IEEE = "ieee"
+ATTR_DATA = "data"
+
+SERVICE_NAMES = {
+    ACTION_SQUAWK: SERVICE_WARNING_DEVICE_SQUAWK,
+    ACTION_WARN: SERVICE_WARNING_DEVICE_WARN,
+}
+
+ACTION_SCHEMA = cv.DEVICE_ACTION_BASE_SCHEMA.extend(
```

Comment:
Add an "action type" here to allow for cleaner code so that we can easily support service calls, cluster method calls etc.